# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/quota/test/xpcshell/test_persist_eviction.js
# Commit: 5ae0cf36c38b
# Full Hash: 5ae0cf36c38bc5d79553382de16f41a2795de86c
# Author: Jan Varga <jvarga@mozilla.com>
# Date: 2024-10-13 21:18:59
# Regressor Bug: 1866402
# File Overlap Count: 1
# Description:
#   Bug 1866402 - Add support for initializing corresponding origin to QuotaManager::OpenClientDirectory; r=devtools-reviewers,nchevobbe
#   
#   QuotaManager::OpenClientDirectory can now get initialization status of
#   corresponding origin and initialize the origin if it's needed.
#   
# ==============================================================================

diff -r 7b3ae7465c92 -r 5ae0cf36c38b dom/quota/test/xpcshell/test_persist_eviction.js
--- a/dom/quota/test/xpcshell/test_persist_eviction.js	Sun Oct 13 10:10:23 2024 +0000
+++ b/dom/quota/test/xpcshell/test_persist_eviction.js	Sun Oct 13 10:10:23 2024 +0000
@@ -60,6 +60,12 @@
       let file = getRelativeFile(path);
       if (index == (persistOldestOrigin ? 2 : 1)) {
         ok(!file.exists(), "The origin directory " + path + " doesn't exist");
+
+        let spec = "http://example" + index + ".com";
+        request = initTemporaryOrigin("default", getPrincipal(spec));
+        await requestFinished(request);
+
+        ok(file.exists(), "The origin directory " + path + " does exist");
       } else {
         ok(file.exists(), "The origin directory " + path + " does exist");
       }
